docs/goals_and_soa.md

```diff
@@ -0,0 +1,392 @@
+# Design principles
+
+## Why Cardano DApp modeling language is important?
+
+While protototypes can be easily constructed with existing Cardano frameworks, making it secure and production ready is sisyphean task.
+
+We believe, that onchain PL improvements by themself could not change that. Only higher-level modeling framework can lead to reliable DApp production.
+
+In following enumerate our high-level design goals
+and demonstrate them by list of specific problems
+arising in existing DApps codebase.
+Such problems are specific security vulnerabilies types
+and various kinds development, audit and support cost increase.
+After that we examine how existing solutions
+are covering our high-level goals.
+
+## High-level goals
+
+1. DApp logic as whole (synced-by-construction)
+2. Code is free from common security weaknesses by construction (secure-by-construction)
+3. Seamplessly emulate and test anything (emulate-anything)
+4. Declarativity close to informal specification and bridging lightweight formal methods (declarative-spec)
+5. Generally production ready (production-ready)
+
+## Not in scope at all
+
+* Composable inter-DApp interactions and DApp migrations
+* Scheduled actions
+
+## Not in scope for current Catalyst funding
+
+* CTL and overall frontend transaction construction model
+* No specific support, but can be added manually
+    * Metadata, reference scripts and non-inlined datums
+    * Stacking and governance features (apart for (security-by-default))
+    * ZK-proofs
+
+# Details of examples of problems to be solved
+
+## Reference apps
+
+Those are list of open-source DApps,
+what we use to demonstrate problems in folloging:
+
+* Audited production DApps
+    * Agora
+    * MinSwap
+    * Fracada
+    * JPG Vesting Contract
+    * Indigo Protocol
+* DApp project we participate, audited or otherwise know it codebase
+    * Hydra Auction
+    * POCRE
+    * CNS
+    * Hydra
+* Plutomicon patterns
+* plutus-usecases
+
+@todo #3: much more links to specific bugs and code size blowups
+
+## On-chain correctness
+
+### Known common vulnerabilities
+
+There is a list of known security vulnerabilities,
+for which naive Plutus implementation is very often prone to.
+Our implementation makes them impossible to happen,
+taking that burden from developers and auditors.
+
+* Double satisfaction
+    * Shown twice in CNS Audit
+    * Shown once in Frada audit (3.2.7)
+* Insufficient staking control
+    * Shown once in CNS Audit - https://github.com/mlabs-haskell/cns-plutus-audit/issues/24
```

Comment:
The link is not accessible for everyone (including me).

Reply:
Yes. I would ask about it later, but I think they are not published code/audit yet.